source/extensions/filters/http/lua/lua_filter.h

```diff
@@ -209,13 +222,50 @@ class StreamHandleWrapper : public Filters::Common::Lua::BaseLuaObject<StreamHan
   DECLARE_LUA_FUNCTION(StreamHandleWrapper, luaLogErr);
   DECLARE_LUA_FUNCTION(StreamHandleWrapper, luaLogCritical);
 
+  /**
+   * Verify cryptographic signatures.
+   * @param 1 (void*)  pointer to public key
+   * @param 2 (string) hash function(including MD4, MD5, SHA1, SHA224, SHA256, SHA384, SHA512,
+   * MD5_SHA1)
+   * @param 3 (string) signature
+   * @param 4 (int)    length of signature
+   * @param 5 (string) clear text
+   * @param 6 (int)    length of clear text
+   * @return (bool, string) If the first element is true, the second element is empty; otherwise,
+   * the second element stores the error message
+   */
+  DECLARE_LUA_FUNCTION(StreamHandleWrapper, luaVerifySignature);
+
+  /**
+   * Decode string encoded in base64.
+   * @param 1 (string) string encoded in base64
+   * @return nil or raw string
+   */
+  DECLARE_LUA_FUNCTION(StreamHandleWrapper, luaDecodeBase64);
+
+  /**
+   * Import public key.
+   * @param 1 (string) keyder string
+   * @param 2 (int)    length of keyder string
+   * @return pointer to public key
+   */
+  DECLARE_LUA_FUNCTION(StreamHandleWrapper, luaImportPublicKey);
```

Comment:
Did you consider using SDS or one of the existing other methods in the API of referencing certificate/key resources? You might be able to avoid import/freeing and resource tracking complexity by doing that.

Reply:
I am not very familiar with the SDS. 

Our customer's case is that they store the public key map (from key id to keyder) in the Lua script and they need a Lua function to import the public key from keyder. I am not sure if SDS or other methods can be applied to this case.